cli/src/semgrep/formatter/text.py

```diff
@@ -491,10 +491,18 @@ def format(
     ) -> str:
         reachable = []
         unreachable = []
-        first_party = []
+        first_party_blocking = []
+        first_party_nonblocking = []
+        first_party_blocking_rules = []
         for match in rule_matches:
             if "sca_info" not in match.extra:
-                first_party.append(match)
+                if match.is_blocking:
+                    first_party_blocking.append(match)
+                    rule_id = match.match.rule_id.value
+                    if rule_id != "-":
```

Comment:
When is the rule_id just a dash? Can you add some comment here to explain it? It feels a little fishy to me

Reply:
Just added a comment! It's when we run semgrep using the -e or --pattern flag and so there is no rule_id specified there and is just set to "-" instead.

Reply:
Ah, perfect! Thanks :)